src/site/content/en/blog/eventing-deepdive/index.md

```diff
@@ -0,0 +1,541 @@
+target. It would result in the following output:
+
+```bash
+"click on document in capturing phase"
+"click on <html> in capturing phase"
+"click on <body> in capturing phase"
+"click on #A in capturing phase"
+"click on #B in capturing phase"
+"click on #C in capturing phase"
+```
+
+{% Aside %} It would be technically more accurate if I had logged "click on #C in target phase".
+However, I chose to use the term "capturing" and "bubbling" for both of `#C`'s event handlers to
+make it clear that handlers can be executed in both phases of the event's lifecycle. Just know that
+technically, the time when the event is at `#C` is officially known as the "target phase".
+{% endAside %}
+
+Note that the event handler for `#C` in which we log "click on #C in the capturing phase" _still_
+executes, but the one in which we log "click on #C in the bubbling phase" does not. This should make
+perfect sense. We called `stopPropagation()` _from_ the former, so that is the point at which the
+event's propagation will cease.
+
+In any of these live demos, I encourage you to play around. Try clicking on the `#A` element only or
+the `body` element only. Try to predict what will happen and then observe if you are correct. At
+this point, you should be able to predict pretty accurately.
+
+### `event.stopImmediatePropagation()`
+
+What is this strange, and not oft-used method? It's similar to stopPropagation, but rather than
+stopping an event from traveling to descendents (capturing) or ancestors (bubbling), this method
+only applies when you have more than one event handler wired up to a single element. Since
+`addEventListener()` supports a multicast style of eventing, it's completely possible to wire up an
+event handler to a single element more than once. When this happens, (in most browsers), event
+handlers are typically executed in the order they were wired up. Calling
+`stopImmediatePropagation()` prevents any subsequent handlers from firing. Consider the following
+example:
+
+```html
+<html>
+  <body>
+    <div id="A">I am the #A element</div>
+  </body>
+</html>
+```
+
+```js
+document.getElementById('A').addEventListener(
+  'click',
+  function (e) {
+    console.log('When #A is clicked, I shall run first!');
+  },
+  false,
+);
+
+document.getElementById('A').addEventListener(
+  'click',
+  function (e) {
+    console.log('When #A is clicked, I shall run second!');
+    e.stopImmediatePropagation();
+  },
+  false,
+);
+
+document.getElementById('A').addEventListener(
+  'click',
+  function (e) {
+    console.log('When #A is clicked, I would have run third, if not for stopImmediatePropagation');
+  },
+  false,
+);
+```
+
+The above example will result in the following console output:
+
+```bash
+"When #A is clicked, I shall run first!"
+"When #A is clicked, I shall run second!"
+```
+
+Note that the third event handler never runs due to the fact that the second event handler calls
+`e.stopImmediatePropagation()`.
```

Comment:
Maybe spell this out more clearly with an extra line:

"If we had instead called `e.stopPropagatino()`, the third handler would still run."